src/package/packageInstall.ts

```diff
@@ -0,0 +1,263 @@
+      delete request.ApexCompileType;
+    }
+  }
+
+  await Lifecycle.getInstance().emit('PackageInstallRequest:presend', request);
+
+  const result = await connection.tooling.create('PackageInstallRequest', request);
+
+  await Lifecycle.getInstance().emit('PackageInstallRequest:postsend', result);
+
+  const packageInstallRequestId = result.id;
+  if (!packageInstallRequestId) {
+    throw installMsgs.createError('packageInstallRequestError', [
+      request.SubscriberPackageVersionKey,
+      result.errors.toString(),
+    ]);
+  }
+
+  if (options?.pollingTimeout == null) {
+    return getStatus(connection, packageInstallRequestId);
+  } else {
+    return pollStatus(connection, packageInstallRequestId, options);
+  }
+}
+
+/**
+ * Returns an array of RSS and CSP external sites for the package.
+ *
+ * @param connection The `Connection` object to the org.
+ * @param subscriberPackageVersionId The ID of the subscriber package version (begins with "04t")
+ * @param installationKey The installation key (if any) for the subscriber package version.
+ * @returns an array of RSS and CSP site URLs, or undefined if the package doesn't have any.
+ */
+export async function getExternalSites(
+  connection: Connection,
+  subscriberPackageVersionId: string,
+  installationKey?: string
+): Promise<Optional<string[]>> {
+  const installKey = escapeInstallationKey(installationKey);
+  const queryNoKey = `SELECT RemoteSiteSettings, CspTrustedSites FROM SubscriberPackageVersion WHERE Id ='${subscriberPackageVersionId}'`;
+  const queryWithKey = `SELECT RemoteSiteSettings, CspTrustedSites FROM SubscriberPackageVersion WHERE Id ='${subscriberPackageVersionId}' AND InstallationKey ='${installKey}'`;
+
+  let queryResult: QueryResult<SubscriberPackageVersion>;
+  try {
+    getLogger().debug(`Checking package: [${subscriberPackageVersionId}] for external sites`);
+    queryResult = await connection.tooling.query<SubscriberPackageVersion>(queryWithKey);
+  } catch (e) {
+    // First check for Implementation Restriction error that is enforced in 214, before it was possible to query
+    // against InstallationKey, otherwise surface the error.
+    if (e instanceof Error && isErrorFromSPVQueryRestriction(e)) {
+      queryResult = await connection.tooling.query<SubscriberPackageVersion>(queryNoKey);
+    } else {
+      throw e;
+    }
+  }
+
+  if (queryResult?.records?.length > 0) {
+    const record = queryResult.records[0];
+    const rssUrls = record.RemoteSiteSettings.settings.map((rss) => rss.url);
+    const cspUrls = record.CspTrustedSites.settings.map((csp) => csp.endpointUrl);
+
+    const sites = [...rssUrls, ...cspUrls];
+    if (sites.length) {
+      return sites;
+    }
+  }
+}
+
+async function getStatus(connection: Connection, installRequestId: string): Promise<PackageInstallRequest> {
+  const result = await connection.tooling.retrieve('PackageInstallRequest', installRequestId);
```

Comment:
I know retrieve doesn't apply the type correctly. maybe replace with singleRecordQuery?

Reply:
Maybe.  Whatever we decide we should be consistent.  I was hoping we could modify jsforce to return proper typing.